multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/validators/parameters/RoutesValidator.java

```diff
@@ -21,15 +20,17 @@
     private final Map<String, ParameterValidator> validators;
     private final Set<String> supportedParamsWithoutValidators;
 
-    public RoutesValidator() {
-        this.validators = new HashMap<>();
-        ParameterValidator routeValidator = new RouteValidator();
-        this.validators.put(routeValidator.getParameterName(), routeValidator);
-
-        this.supportedParamsWithoutValidators = new TreeSet<>();
+    public RoutesValidator(String namespace, boolean applyNamespaceGlobal) {
+        this.validators = getRoutesValidators(namespace, applyNamespaceGlobal);
+		this.supportedParamsWithoutValidators = new TreeSet<>();
         this.supportedParamsWithoutValidators.add(SupportedParameters.NO_HOSTNAME);
     }
 
+    protected Map<String, ParameterValidator> getRoutesValidators(String namespace, boolean applyNamespaceGlobal) {
```

Comment:
maybe rename this to initValidators instead of get, since it should be called more than once in the constructor of an instance; or just put this in the constructor and overwrite in child class

Reply:
If make it as first suggestion, I have to move initialization of validators in method but then I have to remove final modifier. In the second option you cannot override parent constructor because child class always calls parent constructor and then initialize itself.

Reply:
Maybe just rename and either remove final or keep final and setting outside then. Can always keep it as it is, I just find the 'get' word to be confusing in this case.